tombstones.go

```diff
@@ -150,36 +150,45 @@ func readTombstones(dir string) (memTombstones, error) {
 			return nil, d.err()
 		}
 
-		stonesMap.add(k, Interval{mint, maxt})
+		stonesMap.addInterval(k, Interval{mint, maxt})
 	}
 
 	return stonesMap, nil
 }
 
-type memTombstones map[uint64]Intervals
-
-var emptyTombstoneReader = memTombstones{}
+type memTombstones struct {
+	mts map[uint64]Intervals
```

Comment:
`mts` is quite close to `mtx` and a bit confusing
whould you mind changing this to `intvlGroups`